File: trivially/src/main/java/trivially/ep/m7/finalized/Neg.java
package trivially.ep.m7.finalized;

import trivially.ep.m7.Exp;

public class Neg implements trivially.ep.m7.Neg {

    protected Exp inner;

    public Exp getInner() {
        return this.inner;
    }

    // Warning: Unsafe casts
    public void setInner(trivially.ep.m4.Exp inner) { this.inner = (trivially.ep.m7.Exp) inner; }

    public Neg(Exp inner) {
        this.inner = inner;
    }
}

File: coco_csharp/CoCo/EP/M7/Finalized/Neg.cs
namespace EP.M7.Finalized {
    public class Neg : Exp, EP.M7.Neg<Exp> {
        private EP.Exp<Exp> inner;

        public Neg(EP.Exp<Exp> _inner) {
            this.inner = _inner;
        }

        public EP.Exp<Exp> getInner() { return this.inner; }
        public void setInner(EP.Exp<Exp> _inner) { this.inner = _inner; }
    }
}